usage example added in extract glimpse

imransalam opened the pull request:
Added usage example in extract glimpse. Issue raised on https://github.com/tensorflow/tensorflow/issues/30097

imransalam committed:
usage example added in extract glimpse

Added usage example in extract glimpse. Issue raised on https://github.com/tensorflow/tensorflow/issues/30097

rthadur commented on tensorflow/python/ops/image_ops_impl.py:
````diff
@@ -3678,6 +3678,24 @@ def extract_glimpse(
+    
+  Usage Example:
+    ```python
+    >> import tensorflow as tf
````
please remove >> symbol

imransalam replied:
@rthadur , Hey I removed the >> symbol. Thanks

rthadur replied:
thank you

imransalam committed:
Removed the >>

Removed the >> from usage example

alextp commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -3678,6 +3678,22 @@ def extract_glimpse(
+    GLIMPSE_SIZE = (24, 24)
+    image = tf.random.normal(shape=(BATCH_SIZE, IMAGE_HEIGHT, 
+    IMAGE_WIDTH, CHANNELS) )
+    offsets_variable = tf.random.uniform(shape=(BATCH_SIZE, 2))
```
Why offsets_variable? Why not just offsets?

alextp commented on tensorflow/python/ops/image_ops_impl.py:
````diff
@@ -3678,6 +3678,22 @@ def extract_glimpse(
+    
+  Usage Example:
+    ```python
+    import tensorflow as tf
````
No need to import tensorflow

alextp commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -3678,6 +3678,22 @@ def extract_glimpse(
+    offsets_variable = tf.random.uniform(shape=(BATCH_SIZE, 2))
+    output = tf.image.extract_glimpse(image, size=GLIMPSE_SIZE, 
+    offsets=offsets_variable)
+    print(output.shape)
```
It'd be nice if we could show something here at all. If I read this code and not execute it I get almost no information out of it (other than the requirements on the input shapes of the op).

Can we pick a known small one-channel image and show what a glimpse is?

imransalam replied:
Sure let me add that

imransalam committed:
variable name changed import tf removed

Changed offsets_variable to offsets and import tensorflow as tf removed

imransalam committed:
Usable example added in extract_glimpses

Added a visual example of extract_glimpses

alextp commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -3681,17 +3681,24 @@ def extract_glimpse(
+      offsets=[[1, 1]], centered=False, normalized=False)
+    print(image)
+        <tf.Tensor: id=172, shape=(3, 3), dtype=float32, numpy=
+        array([[0.98461604, 0.14595735, 0.3296578 ],
```
Showing particular random numbers is not super helpful. Can you put numbers like 1. 2. 3. etc in the image so a reader has an idea of where each input ended up?

imransalam replied:
Hey I've made the changes, let me know if it's fine or do we need more changes?

imransalam committed:
Random numbers to Sequential

Changed the example usage from random numbers to sequential numbers for easy reading.
